### PR TITLE
[AT-5259] Test manual provisioning

### PR DESCRIPTION
**Pull request by okaycj-atat**
This PR is to add unit testing to exercise the manual provisioning scripts.

[Jira Story AT-5259](https://ccpo.atlassian.net/browse/AT-5259)

### CODE REVIEW

**Comment by okaycj-atat** [comment]
> Also, I noticed that you removed some of the polling logic from these scripts. Aren't some of these jobs async? Did you find that you didn't need to poll in these cases?

@graham-dds I, unfortunately, don't know.  I don't believe there's a way to actually run these scripts using Azure or even Hybrid.  I double check to see if there's a way to handle the polling.  I had assumed the polling was now handled by the function

**Comment by graham-dds** [comment]
> I had assumed the polling was now handled by the function

It is in some places for now (I think only for creating management groups 🤔 ), but generally the idea is that we utilize the state machine (in the case of portfolio provisioning) to track the status of async jobs. For these scripts, I think we will still need to poll.

**Comment by okaycj-atat** [comment]
> It is in some places for now (I think only for creating management groups 🤔 ), but generally the idea is that we utilize the state machine (in the case of portfolio provisioning) to track the status of async jobs. For these scripts, I think we will still need to poll.

Yup, I wrote that and then immediately regretted it.

File: tests/test_manual_provisioning.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    def _helper(self, prov_func, input_path, output_path):
+        csp, inputs = get_provider_and_inputs(input_path, "mock-test")
+        result = prov_func(csp, inputs)
+        update_and_write(inputs, result, output_path, verbose=False)
+        with open(output_path) as json_file:
+            assert json.load(json_file)
+
+    def test_create_tenant(self, tmp_path_factory):
+        tmp_path = tmp_path_factory.getbasetemp()
```
[the remark this answers is not in the document]
Comment (okaycj-atat) [reply tests/test_manual_provisioning.py:24]: That's a better idea.  Thanks.

File: script/provision/b_setup_billing.py [hunk 2]
```diff
@@ -10,20 +9,20 @@
...
         get_billing_profile = BillingProfileVerificationCSPPayload(
             **{**inputs.get("initial_inputs"), **inputs.get("csp_data"), **csp_response}
         )
-        result = csp.create_billing_profile_verification(get_billing_profile)
-        if result.get("status") == "ok":
-            csp_response = result.get("body").dict()
+        try:
+            return csp.create_billing_profile_verification(get_billing_profile)
+        except GeneralCSPException:
+            time.sleep(10)
```
[the remark this answers is not in the document]
Comment (okaycj-atat) [reply script/provision/b_setup_billing.py:24]: This is a great comment.  And will be addressed in https://ccpo.atlassian.net/browse/AT-5358.

**Comment by graham-dds** [comment]
> Since these tests are just supposed to confirm that the scripts can be run end-to-end

Yeah, if we start making assertions about what is in the data at the end of each step or if we run into another February situation where this is what we might actually use to provision portfolios, this testing setup could use another look. For now it's probably fine though.

**Pull request merged**